mmv1/templates/terraform/pre_create/alloydb_instance.go.erb

```diff
@@ -0,0 +1,28 @@
+// Only set the default SSL mode if the user does not explicitly specify it
```

Comment:
Sorry for jumping in and asking a question about this change?
Is there any reason that we need to explicitly set it on the Terraform end if it's default in the server end?

Reply:
No worries. The default SSL mode is not actually set yet on the server end (we are planning to do that soon, and I'll send another PR to update this in terraform once those changes have also been rolled out to prod).

Reply:
I see. While updating whatever set as default in the pre_create is also considered breaking changes and should wait until major release

Reply:
Oh, I did not know that's also considered a breaking change as well. I thought since we set the default in pre_create instead of using `default_value`, then removing the pre_create code but keeping `default_from_api` wouldn't be considered a breaking change. Thanks for letting me know

Reply:
Hello, @nancynh, is it possible to wait until the the default SSL model is set on the server side, and then add this feature in the terraform provider? I mean reverting this PR, and then using `default_from_api` until the server side is ready.

Reply:
It's preferable to keep this feature in. So as an alternative, can we simply remove the changes to explicitly set the SSL mode, and keep the `default_from_api`? So instead SSL mode will default to `ALLOW_UNENCRYPTED_AND_ENCRYPTED_ONLY` temporarily, until we fix it in the server side which at that point will then default to `ENCRYPTED_ONLY`?

Otherwise, we're also okay with just having the default be `ALLOW_UNENCRYPTED_AND_ENCRYPTED_ONLY` until the next major release but the first option is preferable so we don't have to wait as long.

Reply:
@nancynh when the API changes the default value, will that affect existing resources/users?

Reply:
I don't think it should. When a resource is created, its settings (in this case SSL mode) will not change unless the user explicitly sends an update request to change that ssl mode. In other words, unless the resource itself gets destroyed and then recreated (if for example user decides to rename the resource in the configuration file), it should stay the same.

Reply:
wait, is `ENCRYPTED_ONLY` the default that will be changed to?

Reply:
Yes, we're planning on changing default in API to be `ENCRYPTED_ONLY`

Reply:
To be clear, if we get the default value from the API right now it will be `ALLOW_UNENCRYPTED_AND_ENCRYPTED_ONLY`. Though, if a user create the same resource via the UI or gcloud it will be `ENCRYPTED_ONLY`

Reply:
I am going to revert this PR.

Reply:
Talked with the team, you can wait until the default value of SSL mode in server side is fixed. Or you can just use `default_from_api` and not use pre_create and pre_update. Even though the second option is not recommended.

Reply:
Okay, could I get an explanation for why we can't use `default_from_api` only, and then later change the default value in the API? I don't understand how this would be a breaking change (if it is). Because based on the documentation "This attribute is useful for complex or frequently-changed API-side defaults" so it seems like it should be fine if the API-side defaults change?

Reply:
You can use `default_from_api` only, even though it is a breaking change in server side.

Reply:
Sorry I'm still confused. So if we use `default_from_api` only, then would it be okay for TF side? On the server side, we're okay with having user's ssl mode default to `ALLOW_UNENCRYPTED_AND_ENCRYPTED_ONLY` until we change it in the API. We plan on testing and making sure we don't break any existing users before making this change

Reply:
It is OK for terraform to use `default_from_api`.  With `default_from_api`, it sends an empty value to API, so If the default value changed in the server side later, it is not a breaking change on terraform side.